File: InvestmentPortfolio/Models/PortfolioPurchase.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace InvestmentPortfolio.Models
{
    public class PortfolioPurchase
    {
        public int Id { get; set; }
        public int SecurityId { get; set; }
        public int PortfolioId { get; set; }
        public int AmountPurchase { get; set; }
        public DateTime PurchaseDate { get; set; }

        public Security Security { get; set; }

        public Portfolio Portfolio { get; set; }
    }
}

File: InvestmentPortfolio/client/src/components/PortfolioPurchase/PortfolioPurchase.js
import React from "react";
import { Card, CardBody, Button } from "reactstrap";
import { getAllPortfolioPurchases, deletePortfolioPurchase } from "../../modules/portfolioPurchaseManager";

const PortfolioPurchase = ({ portfolioPurchase }) => {
    const handleDelete = (e) => {
        e.preventDefault();
        deletePortfolioPurchase(PortfolioPurchase.id).then(() => {
            getAllPortfolioPurchases()
        });
    };

    return (
        <Card>
            <h3>My Portfolio Purchase</h3>

            <CardBody>
                <p> PortfolioPurchase Id: {portfolioPurchase.id}</p>
                <p> PortfolioPurchase SecurityId: {portfolioPurchase.securityId}</p>
                <p> PortfolioPurchase PortfolioId: {portfolioPurchase.portfolioId}</p>
                <p> PortfolioPurchase AmountPurchase: {portfolioPurchase.amountPurchase}</p>
                <p> PortfolioPurchase PurchaseDate: {portfolioPurchase.purchaseDate}</p>
                <Button className="btn btn-primary" onClick={handleDelete}>Delete</Button>
            </CardBody>
        </Card>
    );
};

export default PortfolioPurchase;
// PortfolioPurchase: Id, SecurityId, PortfolioId, AmountPurchase, PurchaseDate